File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        
        i = 0
        res = ""
        done1, done2 = False, False
        while True:
            try:
                res += word1[i]
            except IndexError:
                done1 = True

            try:
                res += word2[i]
            except IndexError:
                done2 = True

            i += 1

            if done1 and done2:
                return res
        

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder res = new StringBuilder();
        int i = 0;
        while (i < word1.length() && i < word2.length()) {
            res.append(word1.charAt(i));
            res.append(word2.charAt(i));
            i++;
        };

        if (i < word1.length()) {
            res.append(word1.substring(i, word1.length()));
        };
        if (i < word2.length()) {
            res.append(word2.substring(i, word2.length()));
        };

        return res.toString();
    }
}